core/lib/src/config/secret_key.rs

```diff
@@ -178,6 +193,80 @@ impl SecretKey {
     {
         ser.serialize_bytes(&[0; 32][..])
     }
+
+    /// Encrypts the given data.
+    /// Generates a random nonce for each encryption to ensure uniqueness.
+    /// Returns the Vec<u8> of the concatenated nonce and ciphertext.
+    ///
+    /// # Example
+    /// ```rust
+    /// use rocket::config::SecretKey;
+    ///
+    /// let plaintext = "I like turtles".as_bytes();
+    /// let secret_key = SecretKey::generate().unwrap();
+    ///
+    /// let encrypted = secret_key.encrypt(&plaintext).unwrap();
+    /// let decrypted = secret_key.decrypt(&encrypted).unwrap();
+    ///
+    /// assert_eq!(decrypted, plaintext);
+    /// ```
+    pub fn encrypt<T: AsRef<[u8]>>(&self, value: T) -> Result<Vec<u8>, Error> {
+        // Convert the encryption key to a fixed-length array
+        let key: [u8; KEY_LEN] = self.key
+            .encryption()
+            .try_into()
+            .map_err(|_| Error::KeyLengthError)?;
+
+        // Create a new AES-256-GCM instance with the provided key
+        let aead = Aes256Gcm::new(GenericArray::from_slice(&key));
```

Comment:
I think we likely want to use a more a nonce-resistant algorithm here, either AES-GCM-SIV or XChaCha20Poly1305.

Reply:
Great point! Should we do the same for [cookie-rs](https://github.com/rwf2/cookie-rs/blob/master/src/secure/private.rs#L60)?

Reply:
I am going to use `XChaCha20Poly1305` because the crate `chacha20poly1305` has received a [security audit](https://docs.rs/chacha20poly1305/latest/chacha20poly1305/#security-notes), and the crate `aes_gcm_siv` has not been audited [yet](https://docs.rs/aes-gcm-siv/latest/aes_gcm_siv/#security-warning).

Reply:
That sounds good. `cookie-rs` has slightly different requirements. It may be worthwhile to change our algorithm there too, at least to `aes-gcm-siv` but the trade-off is different there. It's something I'm considering, but it's orthogonal to this PR.